plutus-cbor/test/Plutus/Codec/CBOR/EncodingSpec.hs

```diff
@@ -39,27 +47,87 @@ import Test.QuickCheck (
   choose,
   conjoin,
   counterexample,
-  forAll,
+  elements,
   forAllBlind,
+  forAllShrink,
   label,
+  liftShrink2,
   oneof,
+  shrinkList,
+  suchThat,
+  vector,
+  vectorOf,
+  withMaxSuccess,
   (===),
  )
+import qualified Prelude
 
 spec :: Spec
 spec = do
   describe "our CBOR encoding matches oracle's ('cborg' library)" $ do
-    prop "for all (x :: Integer) w/ 'encodeInteger'" $
-      forAll genInteger $
-        propCompareWithOracle CBOR.encodeInteger encodeInteger
+    prop "for all values" $
+      withMaxSuccess 1000 $
+        forAllShrink genSomeValue shrinkSomeValue $ \(SomeValue x _ encode encode') ->
+          propCompareWithOracle encode encode' x
 
   describe "(on-chain) execution cost of CBOR encoding is small" $ do
```

Comment:
See my other comments about the term `small`. I don't think this is really the case and an appropriate name for this property. Also, I don't think this is really a property in the QuickCheck sense: How about asserting the cost of various known sizes in a benchmark, like I did with the `contract-cost` executable? It could be interesting to have all the metrics in a single place and to be able to output tables and graphs on demand when we need them, for all code we are interested in.